realm/realm-library/src/androidTest/java/io/realm/NotificationsTest.java

```diff
@@ -537,9 +470,12 @@ public void onChange(Realm object) {
                     realm.removeAllChangeListeners();
                     realm.addChangeListener(this);
                     realm.addChangeListener(listenerA);
-                    realm.beginTransaction();
-                    realm.commitTransaction();
-
+                    // Async transaction to avoid endless recursion.
+                    realm.executeTransactionAsync(new Realm.Transaction() {
```

Comment:
Yes, this would be a breaking change but even in the current master, commit local transaction in the Realm global change listener without stop condition will be problematic as well. Since for the local commit is always notified by postAtFront which make it calls the change listener forever, any other event get no chance to execute. So IMO, it sounds like a breaking change for user, but it might not be. calling listeners immediately and calling listener in the VERY NEXT event loop have no difference. Except the latter one wastes a few cpu cycles on event fetching, and the latter one has a chance to execute the code after `commitTransaction` call. None of those can be thought as good practice.

But, well, the code after `commitTransaction()` won't be called is a visible breaking change for user :(